addon/model.js

```diff
@@ -102,41 +103,50 @@ const retrieveFromCurrentState = computed('_topModel.currentState', function(key
   return this._topModel._internalModel.currentState[key];
 }).readOnly();
 
-// global buffer for initial properties to work around
-//  a)  can't write to `this` before `super`
-//  b)  core_object writes properties before calling `init`; this means that no
-//      CP or setknownProperty can rely on any initialization
-let initProperites = Object.create(null);
+const INIT_KEYS = {
+  store: 1,
+  _internalModel: 1,
+  _topModel: 1,
+  _parentModel: 1,
+  isError: 1,
+};
 
-export default class MegamorphicModel extends EmberObject {
+export default class MegamorphicModel {
+  constructor(properties) {
+    this.init(properties);
+  }
   init(properties) {
```

Comment:
do we ever reinvoke `init` ? if not, we may want to inline.

Reply:
We don't, but we do stub it in tests currently